src/connectors/metamask.ts

```diff
@@ -38,46 +39,45 @@ export default class MetaMaskConnector extends CKBConnector {
     });
   }
 
-//   private setAddress(ethAddress: `0x${string}` | undefined) {
-//     if (!ethAddress) {
-//       this.setData(defaultWalletValue);
-//       return;
-//     }
-//     config.initializeConfig(config.predefined.AGGRON4);
-//     const lock = commons.omnilock.createOmnilockScript({
-//       auth: { flag: 'ETHEREUM', content: ethAddress ?? '0x' },
-//     });
-//     const address = helpers.encodeToAddress(lock, {
-//       config: config.predefined.AGGRON4,
-//     });
-//     this.setData({
-//       address,
-//       connectorType: this.type.toLowerCase(),
-//       data: ethAddress,
-//     });
-//   }
+  private setAddress(ethAddress: `0x${string}` | undefined) {
+    if (!ethAddress) {
+      return;
+    }
+    config.initializeConfig(config.predefined.AGGRON4);
+    const lock = commons.omnilock.createOmnilockScript({
+      auth: { flag: 'ETHEREUM', content: ethAddress ?? '0x' },
+    });
+    const address = helpers.encodeToAddress(lock, {
+      config: config.predefined.AGGRON4,
+    });
+    this.store.dispatch(setWallet({
+      address,
+      walletType: 'MetaMask',
+      ethAddress: ethAddress,
+    }));
+  }
 
   public async connect() {
     const { account } = await connect({ connector: new InjectedConnector() });
-    // this.setAddress(account);
-    // this.isConnected = true;
-    // this.listeners.push(
-    //   watchAccount((account) => {
-    //     if (account.isConnected) {
-    //       this.setAddress(account.address);
-    //     }
-    //     if (account.isDisconnected) {
-    //       this.setAddress(undefined);
-    //     }
-    //   }),
-    // );
+    this.setAddress(account);
```

Comment:
- If I logged into a JoyID account, log out, and log into metamask, my gifts created in JoyID will display on the screen until I refresh the page. (this didn't happen if I switch account from metamask to joyid)
- Need to refetch the gift list once user switch account?